internal/services/mssql/mssql_job_target_group_resource.go

```diff
@@ -0,0 +1,327 @@
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			targets, err := expandJobTargets(model.JobTargets)
+			if err != nil {
+				return err
+			}
+
+			parameters := jobtargetgroups.JobTargetGroup{
+				Name: pointer.To(model.Name),
+				Properties: pointer.To(jobtargetgroups.JobTargetGroupProperties{
+					Members: targets,
+				}),
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id, parameters); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r MsSqlJobTargetGroupResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MSSQL.JobTargetGroupsClient
+
+			id, err := jobtargetgroups.ParseTargetGroupID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			state := MsSqlJobTargetGroupResourceModel{
+				Name:       id.TargetGroupName,
+				JobAgentID: jobtargetgroups.NewJobAgentID(id.SubscriptionId, id.ResourceGroupName, id.ServerName, id.JobAgentName).ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				if props := model.Properties; props != nil {
+					state.JobTargets = flattenJobTargets(props.Members)
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r MsSqlJobTargetGroupResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MSSQL.JobTargetGroupsClient
+
+			id, err := jobtargetgroups.ParseTargetGroupID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var config MsSqlJobTargetGroupResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			if existing.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", id)
+			}
+
+			if existing.Model.Properties == nil {
+				return fmt.Errorf("retrieving %s: `model.Properties` was nil", id)
+			}
+
+			if metadata.ResourceData.HasChange("job_target") {
+				targets, err := expandJobTargets(config.JobTargets)
+				if err != nil {
+					return err
+				}
+				existing.Model.Properties.Members = targets
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, *id, *existing.Model); err != nil {
+				return fmt.Errorf("updating: %s: %+v", id, err)
+			}
```

Comment:
This is minor, and I leave it up to you to decide what is appropriate here, after nil checking the returned model and model.Properties we assign it to a variable e.g. `payload` which we patch the changes into.

It's not a big deal here since we only need to reference `existing.Model.Properties` once as opposed to dozens of times like in other resources, but if we wanted to be consistent with the docs and other resources this would look like the snippet below
```suggestion
			payload := existing.Model
			
			if metadata.ResourceData.HasChange("job_target") {
				targets, err := expandJobTargets(config.JobTargets)
				if err != nil {
					return fmt.Errorf("expanding `job_target`: %+v", err)
				}
				payload.Properties.Members = targets
			}

			if _, err := client.CreateOrUpdate(ctx, *id, *payload); err != nil {
				return fmt.Errorf("updating: %s: %+v", id, err)
			}
```